test/new-e2e/tests/containers/docker/docker_test.go

```diff
@@ -0,0 +1,141 @@
+	// Full Agent
+	suite.Run(t, &DockerSuite{stackName: "docker-stack", isDsdStandalone: false})
+
+	// DogstatsD Standalone
+	suite.Run(t, &DockerSuite{stackName: "docker-stack", isDsdStandalone: true})
+}
+
+func (suite *DockerSuite) getAgentImage() string {
+	if suite.isDsdStandalone {
+		return "datadog/dogstatsd:latest"
+	}
+
+	return ""
+}
+
+func (suite *DockerSuite) SetupSuite() {
+
+	ctx := context.Background()
+
+	stackConfig := runner.ConfigMap{
+		"ddagent:deploy":        auto.ConfigValue{Value: "true"},
+		"ddagent:fakeintake":    auto.ConfigValue{Value: "true"},
+		"ddagent:fullImagePath": auto.ConfigValue{Value: suite.getAgentImage()},
+	}
+
+	if suite.isDsdStandalone {
+		stackConfig["ddagent:fullImagePath"] = auto.ConfigValue{Value: "datadog/dogstatsd:latest"}
+	}
+
+	_, stackOutput, err := infra.GetStackManager().GetStackNoDeleteOnFailure(ctx, suite.stackName, stackConfig, dockervm.Run, false)
+
+	if !suite.Assert().NoError(err) {
+		suite.FailNow(ctx)
+	}
+
+	vmConnection := stackOutput.Outputs["vm-connection"].Value.(map[string]interface{})
+	suite.stackName = fmt.Sprintf("%s-%v", os.Getenv("USER"), vmConnection["host"])
+
+	fakeintakeHost := stackOutput.Outputs["fakeintake-host"].Value
+	suite.Fakeintake = fakeintake.NewClient(fmt.Sprintf("http://%s", fakeintakeHost))
+
+	suite.Eventuallyf(
+		func() bool { return suite.Fakeintake.GetServerHealth() == nil },
+		1*time.Minute,
+		5*time.Second,
+		"Fakeintake is never healthy",
+	)
+
+	err = suite.Fakeintake.FlushServerAndResetAggregators()
+	if !suite.Assert().NoError(err) {
+		fmt.Println("Failed to flush/reset fakeintake")
+		suite.FailNow(ctx)
+	}
+
+	suite.BaseSuite.SetupSuite()
+}
+
+func (suite *DockerSuite) FailNow(ctx context.Context) {
+	_, err := infra.GetStackManager().GetPulumiStackName(suite.stackName)
+	suite.Require().NoError(err)
+	if !runner.GetProfile().AllowDevMode() || !*containers.KeepStacks {
+		infra.GetStackManager().DeleteStack(ctx, suite.stackName)
+	}
+	suite.T().FailNow()
+}
+
+func (suite *DockerSuite) TearDownSuite() {
+	suite.BaseSuite.TearDownSuite()
+}
+
+func (suite *DockerSuite) TestDSDWithUDS() {
+	suite.AssertMetric(&containers.TestMetricArgs{
+		Filter: containers.TestMetricFilterArgs{
+			Name: "custom.metric.uds",
+		},
+		Expect: containers.TestMetricExpectArgs{
+			Tags: &[]string{
+				`^container_name:metric-sender-uds$`,
+				`^short_image`,
+				`^image_name`,
+				`^image_id`,
+				`^docker_image`,
+				`^container_id`,
+			},
+		},
+	})
+}
+
+func (suite *DockerSuite) TestDSDWithUDP() {
+	suite.AssertMetric(&containers.TestMetricArgs{
+		Filter: containers.TestMetricFilterArgs{
+			Name: "custom.metric",
+		},
+
+		Expect: containers.TestMetricExpectArgs{
+			Tags: &[]string{
+				`^container_name:metric-sender-udp$`,
+				`^short_image`,
+				`^image_tag`,
+				`^docker_image`,
+				`^git.commit.sha`,
+				`^git.repository_url`,
+				`^series`,
+				`^image_name`,
+				`^image_id`,
+				`^container_id`,
```

Comment:
Why UDP version has more tags than UDS?

Reply:
- The `series` tag is coming from the app I am using to send metrics over UDP (defined [here](https://github.com/DataDog/test-infra-definitions/tree/main/components/datadog/apps/dogstatsd/images/dogstatsd))
- The `image_tag` tag is missing on UDS because the image used in the container is specified as `ubuntu`. Setting it to `ubuntu:latest` results in having `image_tag:latest` in the metric tags. (So I added this to be able to assert this tag)
- The remaining missing tags in case of UDS are `git.commit.sha` and `git.repository_url`. I will investigate why these are missing. 

Thanks

Reply:
I am now using the same app to emit metrics on both udp and uds, so all tags are now being asserted